addons/purchase/models/purchase_order.py

```diff
@@ -185,7 +185,9 @@ def _compute_currency_rate(self):
     def _compute_date_planned(self):
         """ date_planned = the earliest date_planned across all order lines. """
         for order in self:
-            dates_list = order.order_line.filtered(lambda x: not x.display_type and x.date_planned).mapped('date_planned')
+            relevant_lines = order.order_line.filtered(lambda x: not x.display_type and x.date_planned)
+            relevant_lines = relevant_lines.filtered(lambda x: x._origin) or relevant_lines
```

Comment:
I think this should be equivalent to 

```suggestion
            relevant_lines = relevant_lines._origin or relevant_lines
```